src/kontrol/prove.py

```diff
@@ -316,7 +318,40 @@ def init_and_run_proof(test: FoundryTest) -> APRFailureInfo | Exception | None:
             start_server=start_server,
             port=options.port,
             maude_port=options.maude_port,
+            haskell_threads=options.max_frontier_parallel,
         ) as kcfg_explore:
+
+            def create_kcfg_explore() -> KCFGExplore:
+                if options.maude_port is None:
+                    dispatch = None
+                else:
+                    dispatch = {
+                        'execute': [('localhost', options.maude_port, TransportType.HTTP)],
+                        'simplify': [('localhost', options.maude_port, TransportType.HTTP)],
+                        'add-module': [
+                            ('localhost', options.maude_port, TransportType.HTTP),
+                            ('localhost', kcfg_explore.cterm_symbolic._kore_client.port, TransportType.SINGLE_SOCKET),
+                        ],
+                    }
+                client = KoreClient(
+                    'localhost',
+                    kcfg_explore.cterm_symbolic._kore_client.port,
```

Comment:
This is a roundabout way of getting the server port. Can you think of a way to improve this?

Reply:
It's not easy with `legacy_explore` from what I know, since that yields the `KCFGExplore` from inside the `with kore_server(...) as` block, and not the server itself, but I've refactored it so we start the server (if necessary) directly in `init_and_run_proof` and initialize the `KCFGExplore` used for initializing the proof with `create_kcfg_explore()` which I already had here.